### PR TITLE
Add authentification

### PR DESCRIPTION
**Pull request by notFloran**
Setup database with Flask-SQLAlchemy
Setup api authentification with Flask-JWT
Add User model

### CODE REVIEW

File: rushapi/rush.py [hunk 1]
```diff
@@ -1,14 +1,30 @@
 from flask import Flask
 from flask_restful import Resource, Api
+from flask_jwt import JWT, jwt_required
+from flask.ext.sqlalchemy import SQLAlchemy
+from models import User
 
 app = Flask(__name__)
+app.config['SECRET_KEY'] = 'super-secret'
```
Comment (martin-damien) [inline rushapi/rush.py]: We will need to load those settings from a configuration file :

``` bash
$ rush -c config.yml
```

**Comment by notFloran** [comment]
I setup structure like explained here : https://flask-restful.readthedocs.org/en/latest/intermediate-usage.html#project-structure

Now to run the app : ```python run.py```

I need help for the import of User in app.py, if I move the import to the top, I can't start the app.

**Comment by martin-damien** [comment]
Please add documentation on how to connect:

- Wich headers are required
- Give an example with a Postman screenshot for exemple

File: rushapi/models.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    id = db.Column(db.Integer, primary_key=True)
+    created_at = db.Column(db.DateTime,  default=db.func.current_timestamp())
+    updated_at = db.Column(db.DateTime,  default=db.func.current_timestamp(), onupdate=db.func.current_timestamp())
+
+
+class User(Base):
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(80), unique=True)
+    _password = db.Column('password', db.String(128))
```
Comment (martin-damien) [inline rushapi/models.py]: Why is there an underscore?

File: rushapi/models.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    email = db.Column(db.String(120), unique=True)
+
+    def __init__(self, username, password, email):
+        self.username = username
+        self.password = password
```
Comment (martin-damien) [inline rushapi/models.py]: Missing underscore?

File: run.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+from rushapi.app import app
```
Comment (martin-damien) [inline run.py]: run.py will be out of scope when deploying. I'm not sure that the application should be here.

- You do nothing more than calling the real application entry point
- You'll have nothing to test
- setup.py will create a shortcut to the method you want
- You can setup the IDE to run the file you are calling

**Pull request merged**